src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -478,6 +479,12 @@ private void VisitEquals(BinaryExpression expression)
             }
 
             var memberInfo = GetMember(expression.Left);
+            
+            if (memberInfo.Path == string.Empty)
+            {
+                throw new InvalidQueryException("You cannot compare the whole document in Where closure. Please write a conditional statement using fields from the document. A query such as session.Query<User>().Where(u => u == null) is not meaningful, you are asserting that the document itself is not null, which it can never be. Did you intend to assert that a property isn't set to null?");
```

Comment:
Can we move this message to a const? Make it internal so you'll be able assert with the usage of it in the tests